MySQL/Scripts/Script.sql

```diff
@@ -0,0 +1,279 @@
+VALUES 
+	('nothing happened', 102, CURRENT_TIMESTAMP(), 'edit');
+
+-- CREATE some Views
+
+CREATE VIEW v_get_products_data AS
+	SELECT p.id, p.name, p.shortDescription, p.weight, p.price, p.barcodeNumber 
+		FROM product AS p;
+
+SELECT * FROM v_get_products_data; 
+
+CREATE VIEW v_get_customers_data AS
+	SELECT c.id, c.name, c.financialId, c.vatNumber, c.address, c.contactPerson 
+		FROM customer AS c;
+
+SELECT * FROM v_get_customers_data; 
+
+-- CREATE  search queries
+
+DROP PROCEDURE IF EXISTS sp_get_product_info_by_part_product_name;
+
+DELIMITER $$
+CREATE PROCEDURE sp_get_product_info_by_part_product_name
+(
+   productName VARCHAR(255)
+) 
+BEGIN 
+   SELECT * FROM get_products_data 
+   WHERE name LIKE CONCAT('%', productName, '%'); 
+end$$
+DELIMITER;
+
+CALL get_product_info_by_part_product_name('hai'); 
+
+-- DROP PROCEDURE sp_get_product_info_by_part_product_name;
+
+CREATE PROCEDURE sp_get_order_info_by_order_number
+(
+   orderNumber INT
+) 
+BEGIN  
+	SELECT po.orderId, c.vatNumber, c.name, p.name, po.productQuantity, p.price AS 'price per unit', p.price * po.productQuantity AS 'total price' 
+		FROM productOrder AS po
+		JOIN product AS p
+		ON p.id = po.productId 
+		JOIN customerOrder AS o
+		ON o.id = po.orderId 
+		JOIN customer AS c
+		ON c.id = o.customerId
+		WHERE po.orderId = orderNumber;
+END; 
+
+CALL sp_get_order_info_by_order_number(101);
+  
+-- DROP PROCEDURE sp_get_order_info_by_orderNumber;
+
+CREATE PROCEDURE sp_get_orders_for_last_month_by_customer_name
+(
+   customerName VARCHAR(255)
+) 
+BEGIN
+	SELECT o.id AS 'ORDER id', o.totalPrice, o.orderDate, c.name AS 'customer name'
+		FROM customerOrder AS o 
+		JOIN customer AS c
+		ON c.id = o.customerId
+		WHERE c.name LIKE CONCAT('%', customerName, '%') 
+		AND DATE_ADD(o.orderDate, INTERVAL 1 MONTH) > current_date();
+END;
+
+CALL sp_get_orders_for_last_month_by_customer_name('Flower');
+
+-- DROP PROCEDURE sp_get_orders_for_last_month_by_customer_name;
+
+CREATE VIEW sp_get_products_sold_last_month AS
+	SELECT p.name AS 'product.name', sum(po.productQuantity) AS 'total quantity sold last month' 
+		FROM productOrder AS po 
+		JOIN product AS p
+		ON po.productId = p.id 
+		JOIN customerOrder AS o
+		ON o.id = po.orderId 
+		WHERE DATE_ADD(o.orderDate, INTERVAL 1 Month) > current_date()
+		GROUP BY p.id 
+		ORDER BY p.name;
+
+SELECT * FROM sp_get_products_sold_last_month; 
+
+-- DROP view sp_get_products_sold_last_month;
+
+-- add indices
+
+CREATE INDEX index_edited_product_log
+ON orderActivityLog (editedProductId);
+
+CREATE INDEX index_edited_order_log
+ON orderActivityLog (editedOrderId);
+
+CREATE INDEX index_edited_customer_log
+ON customerActivityLog (editedCustomerId);
+
+CREATE INDEX index_customer_order
+ON customerOrder (customerID);
+
+-- Log info in logs table
+
+CREATE PROCEDURE sp_insert_in_product_log_table(eventType VARCHAR(255), msg TEXT, productId INT)
```

Comment:
If you only the procedure is for inserting it is good to not provide eventType from outside. 
Someone could provide DELETE/UPDATE as an event.

Reply:
Not valid comment. It was my bad.